src/lib/components/Dropdown/Dropdown.tsx

```diff
@@ -75,6 +75,17 @@ const DropdownComponent: FC<DropdownProps> = ({
   }, [placement]);
 
   const [closeRequestKey, setCloseRequestKey] = useState<string | undefined>(undefined);
+  const wrapperRef = useRef<HTMLDivElement>(null);
+
+  const handleOutsideClick = useCallback(
+    (event: MouseEvent) => {
+      if (dismissOnClick) return;
+      if (wrapperRef.current && !wrapperRef.current.contains(event.target as Node)) {
+        setCloseRequestKey(uuid());
```

Comment:
replace by react 18 useId

Reply:
hi, can't use useid cause will need a new id every time so that component can re-render. if we use useid, it'll work for the first time, but since the id is saved when will repeat the scenario, the component won't render cause of the same id. other changes are doable.